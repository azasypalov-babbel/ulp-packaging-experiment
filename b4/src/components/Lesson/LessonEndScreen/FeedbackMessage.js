import React from 'react';
import PropTypes from 'prop-types';
import cx from '../../../lib/cxHelper';
import { getModifier, getMessage } from './helper';

const FeedbackMessage = ({
  displayName,
  learnLanguageAlpha3,
  grade,
  feedbackMessageText
}) => {
  const prefix = 'lesson-end-screen-feedback-message';
  const scoreModifier = getModifier(prefix, grade);
  const message = getMessage({ displayName, feedbackMessageText });

  const classNames = {
    base: cx(prefix, scoreModifier),
    icon: cx(`${prefix}__icon`),
    text: cx(`${prefix}__text`)
  };
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const imageUrl = require(`@assets/images/feedbackMessage/${learnLanguageAlpha3}/${grade}.svg`);
  return (
    <div className={classNames.base}>
      <img src={imageUrl} className={classNames.icon}/>
      <h1 className={classNames.text} data-feedback-grade={grade}>{message}</h1>
    </div>
  );
};

FeedbackMessage.propTypes = {
  displayName: PropTypes.string.isRequired,
  learnLanguageAlpha3: PropTypes.string.isRequired,
  grade: PropTypes.string.isRequired,
  feedbackMessageText: PropTypes.string.isRequired
};

export default FeedbackMessage;
