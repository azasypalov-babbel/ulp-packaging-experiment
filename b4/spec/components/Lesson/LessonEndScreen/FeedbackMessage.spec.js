import React from 'react';
import FeedbackMessage from '../../../../src/components/Lesson/LessonEndScreen/FeedbackMessage';
import { shallow } from 'enzyme';

const defaultProps = {
  displayName: 'Christian',
  learnLanguageAlpha3: 'FRA',
  grade: 'high',
  feedbackMessageText: `Fantastique, %{display_name} !`
};

describe('<FeedbackMessage />', () => {
  test('it renders', () => {
    const component = shallow(<FeedbackMessage {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
