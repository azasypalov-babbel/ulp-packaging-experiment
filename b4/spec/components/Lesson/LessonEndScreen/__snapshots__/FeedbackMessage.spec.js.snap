// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedbackMessage /> it renders 1`] = `
<div
  className="loy-lesson-end-screen-feedback-message loy-lesson-end-screen-feedback-message--high"
>
  <img
    className="loy-lesson-end-screen-feedback-message__icon"
    src="test-file-stub"
  />
  <h1
    className="loy-lesson-end-screen-feedback-message__text"
    data-feedback-grade="high"
  >
    <span
      key="1"
    >
      Fantastique, 
    </span>
    <br
      key="2"
    />
    <span
      key="3"
    >
      Christian
       !
    </span>
  </h1>
</div>
`;
